neonvm/controllers/virtualmachine_controller.go

```diff
@@ -503,9 +503,68 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 		// TODO: implement RestartPolicy
 		switch virtualmachine.Spec.RestartPolicy {
 		case vmv1.RestartPolicyAlways:
-			// TODO: restart VirtualMachine
+			log.Info("Restarting VM runner pod", "VM.Phase", virtualmachine.Status.Phase, "RestartPolicy", virtualmachine.Spec.RestartPolicy)
+			// get runner to delete
+			vmRunner := &corev1.Pod{}
+			err := r.Get(ctx, types.NamespacedName{Name: virtualmachine.Status.PodName, Namespace: virtualmachine.Namespace}, vmRunner)
+			if err == nil {
+				// delete current runner
+				if err = r.Delete(ctx, vmRunner); err != nil {
+					log.Error(err, "Failed to delete runner Pod", "Pod.Namespace", vmRunner.Namespace, "Pod.Name", vmRunner.Name)
+					return err
+				}
+				r.Recorder.Event(virtualmachine, "Normal", "Deleted",
+					fmt.Sprintf("VM runner pod was deleted: %s", vmRunner.Name))
+				log.Info("VM runner pod was deleted", "Pod.Namespace", vmRunner.Namespace, "Pod.Name", vmRunner.Name)
+				// TODO: remove thos hack after redesign reconcile loop
+				time.Sleep(time.Second)
+			} else if !apierrors.IsNotFound(err) {
+				log.Error(err, "Failed to get runner Pod to delete")
+				return err
+			}
+			// do cleanup
+			virtualmachine.Status.PodName = ""
+			virtualmachine.Status.PodIP = ""
+			virtualmachine.Status.Node = ""
+			virtualmachine.Status.CPUs = nil
+			virtualmachine.Status.MemorySize = nil
+			// drop VM phase
+			virtualmachine.Status.Phase = ""
```

Comment:
we have this code in two places. can we put it into a method somewhere?

Reply:
I originally decided not to add another method to the VirtualMachine type because cleanup only happens in those two places, but now I've added Cleanup().